mne/decoding/receptive_field.py

```diff
@@ -0,0 +1,222 @@
+import numpy as np
+from .base import get_coefs, BaseEstimator, _check_estimator
+
+
+class ReceptiveField(BaseEstimator):
+    """Fit a receptive field model.
+
+    This allows you to fit a model using time-lagged input features. For
+    example, a spectro- or spatio-temporal receptive field (STRF).
+
+    Parameters
+    ----------
+    tmin : int | float
+        The tmining lag. Negative values correspond to times in the past.
+    tmax : int | float
+        The ending lag. Positive values correspond to times in the future.
+        Must be >= tmin.
+    sfreq : int | float
+        The sampling frequency used to convert times into samples.
+    feature_names : array, shape (n_features,) | None
+        Names for input features to the model. If None, feature names will
+        be auto-generated from the shape of input data after running `fit`.
+    model : instance of sklearn model | string | None
+        The model used in fitting inputs and outputs. This can be any
+        sklearn-style model that contains a fit and predict method. Currently
+        the only supported string is 'ridge', which uses the default
+        :mod:`sklearn` ridge model. This is also what `None` defaults to.
+
+    Attributes
+    ----------
+    coef_ : array, shape (n_features, n_delays)
+        The coefficients from the model fit, reshaped for easy visualization.
+        If you want the raw (2d) coefficients, use `mne.decoding.get_coefs`
+        on `self.model`.
+    mask_pred : array, dtype bool, shape (n_times,)
+        A mask with True values corresponding to datapoints that were used
+        in generating model predictions. Creating time delays necessitates that
+        datapoints at the beginning / end of the input are not used.
```

Comment:
not sure what you meant by "ok" but I am assuming it means I don't have to change anything :)